File: backend-ms1/backend-ms1/src/main/java/ma/zs/emailling/service/facade/admin/commun/UtilisateurAdminService.java
package ma.zs.emailling.service.facade.admin.commun;

import java.util.List;
import ma.zs.emailling.bean.core.commun.Utilisateur;
import ma.zs.emailling.dao.criteria.core.commun.UtilisateurCriteria;
import ma.zs.emailling.zynerator.service.IService;



public interface UtilisateurAdminService extends  IService<Utilisateur,UtilisateurCriteria>  {
    Utilisateur findByUsername(String username);
    boolean changePassword(String username, String newPassword);





}

File: frontend/src/app/shared/service/admin/commun/UtilisateurAdmin.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {BehaviorSubject, Observable} from 'rxjs';

import {environment} from 'src/environments/environment';

import {UtilisateurDto} from 'src/app/shared/model/commun/Utilisateur.model';
import {UtilisateurCriteria} from 'src/app/shared/criteria/commun/UtilisateurCriteria.model';
import {AbstractService} from 'src/app/zynerator/service/AbstractService';


@Injectable({
  providedIn: 'root'
})
export class UtilisateurAdminService extends AbstractService<UtilisateurDto, UtilisateurCriteria> {
     constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    public constrcutDto(): UtilisateurDto {
        return new UtilisateurDto();
    }

    public constrcutCriteria(): UtilisateurCriteria {
        return new UtilisateurCriteria();
    }

    get API() {
        return environment.apiUrlEmaillingservice + 'admin/utilisateur/';
    }
}
